### PR TITLE
ref(hc): Replace OrganizationService.get_organizations

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Avoid the problem of resolving to a region by not dispatching to a region silo service in the first place. This follows the comment on get_organization about reading from the control silo's OrganizationMapping table.

To strictly maintain the existing method's functionality, it's still necessary to replace it with HC service calls. In case it was being queried from a region silo, it wouldn't suffice to just look up organizations in that region. The new methods would make an RPC to the control silo and return data from organizations on other regions, if applicable, as looked up in the control silo's mapping table.

To support this, modify RpcOrganizationMapping to extend RpcOrganizationSummary as a drop-in replacement for the original `get_organizations` return value.

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/user/impl.py
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/user/impl.py:113]: As noted inline, we'll probably need to modify the `OrganizationMapping` schema in order to support the `only_visible` parameter. Which would also mean backfilling it and adding functionality to push updates to the outbox. The stopgap solution (reading from `Organization`) should mean that this *is* safe to push to production but *won't* let us add `stable=True` to affected tests yet.

File: src/sentry/services/hybrid_cloud/user/service.py
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/user/service.py:97]: This could just as easily go in `OrganizationMappingService`, but IMHO `UserService` is the second-most intuitive place to go looking for this (first being `OrganizationService`).

From `user_service.get_organizations`, it seems like it should be clear enough that you're getting "the user's organizations", i.e., the ones of which the user is a member. I considered naming it "get_organizations_with_membership" or something, but that invites confusion about whether you're getting back the member object or the org object. (The implementation actually queries for both. It wouldn't be too hard to return a list of `Tuple[RpcOrganizationSummary, RpcOrganizationMemberMapping]` if that would be useful.)

File: src/sentry/services/hybrid_cloud/organization_mapping/model.py
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/organization_mapping/model.py:17]: Note that this is a potentially breaking change: `organization_id` is renamed to `id` with the same meaning. I didn't find any places that `organization_id` was referenced, however. (We'll see if CI turns one up.)

I don't love that `id` could be confused for the primary key of the `OrganizationMapping` table (though we weren't exposing that before and shouldn't start). I think it's worth it to be able to up-cast `RpcOrganizationMapping` to `RpcOrganizationSummary`. We should be encouraging the caller to think of both classes as a miniature slice of an `RpcOrganization`, so `id` should mean the org's ID in all contexts.

File: src/sentry/services/hybrid_cloud/organization/service.py
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/organization/service.py:73]: Deleting this temporarily to make sure nothing breaks. I'll revert dbadfc9 before deploying, for compatibility with `getsentry`. See https://github.com/getsentry/getsentry/pull/10594

**Comment by RyanSkonnord** [comment]
As far as I could tell, the `scope` argument was just never used. Have I missed something? Or was there an intent to use it later, or a previous use that's been deleted? If we do need it, that's something else we'd have to push up into the mapping table along with `status`, and probably messier than `status`.

**Comment by RyanSkonnord** [comment]
One more detail that I wanted to call out explicitly... this is proceeding from the assumption that all the current callers can get by with only the data in `RpcOrganizationSummary`, as the preexisting return type on `OrganizationService.get_organizations` implies. So this is distinct from the problem where we might need full organization metadata that can only be gotten from the region silos, and would therefore need to "fan out" to more than one region.

If and when we encounter a case where "fanning out" is unavoidable, I intend to stick to our plan of forcing the call site to make the multiple service calls itself, which may entail adding an RPC signature along the lines of

```python
def get_all_organizations_for_user_on_region(self, *, region: Region, user_id: int) -> List[RpcOrganization]
```

**Comment by RyanSkonnord** [comment]
Scaled back the scope of this change a bit, such that it's no longer deleting `OrganizationService.get_organizations` but replaces it wherever `only_visible` isn't needed. After this is merged, we can create a new branch to go the rest of the way, by reverting [8b4a09d](https://github.com/getsentry/sentry/pull/49387/commits/8b4a09de5ef7940ab44c7bad110dd8153a3142b5) basically.

**Comment by RyanSkonnord** [comment]
Merging is still blocked on `OrganizationMapping` backfill and synchronization.

File: src/sentry/services/hybrid_cloud/organization_mapping/model.py
Comment (corps) [reply src/sentry/services/hybrid_cloud/organization_mapping/model.py:17]: That seems fair.

File: src/sentry/services/hybrid_cloud/user/service.py
Comment (corps) [reply src/sentry/services/hybrid_cloud/user/service.py:92]: 👍

File: src/sentry/services/hybrid_cloud/organization_mapping/model.py
Comment (markstory) [reply src/sentry/services/hybrid_cloud/organization_mapping/model.py:17]: > We should be encouraging the caller to think of both classes as a miniature slice of an RpcOrganization, so id should mean the org's ID in all contexts.

I agree here. The id of a mapping row isn't something we should need to reference directly, especially once `organization_id` becomes a unique key on the mapping table.

File: src/sentry/services/hybrid_cloud/user/impl.py
Comment (markstory) [inline src/sentry/services/hybrid_cloud/user/impl.py:98]: If we find ourselves doing this kind of query often we might want to add a relationship between OrganizationMemberMapping and OrganizationMapping so that we can do joined loads. Currently there are users with 500+ organization memberships so there could be some benefits from joins.

File: src/sentry/services/hybrid_cloud/organization_mapping/serial.py
Comment (markstory) [inline src/sentry/services/hybrid_cloud/organization_mapping/serial.py:29]: Aren't we removing the `verified` attribute soon @GabeVillalobos ?

Comment (GabeVillalobos) [reply src/sentry/services/hybrid_cloud/organization_mapping/serial.py:29]: We are indeed. Right now we're always defining it as false per the model default and shouldn't be setting it anywhere.

Comment (RyanSkonnord) [reply src/sentry/services/hybrid_cloud/organization_mapping/serial.py:29]: From the mapping you mean? So there's no harm in just proactively removing it from the serialize function here?

**Comment by RyanSkonnord** [comment]
> 🚨 **Warning:** This pull request contains Frontend and Backend changes!
> 
> It's discouraged to make changes to Sentry's Frontend and Backend in a single pull request. The Frontend and Backend are **not** atomically deployed. If the changes are interdependent of each other, they **must** be separated into two pull requests and be made forward or backwards compatible, such that the Backend or Frontend can be safely deployed independently.
> 
> Have questions? Please ask in the [`#discuss-dev-infra` channel](https://app.slack.com/client/T024ZCV9U/CTJL7358X).

I believe this is in reference to [this whitespace change](https://github.com/getsentry/sentry/pull/49387/files#diff-0932c99454a7d5cff4b7bfc483959d40aa6107fccb18f35d363a437633336bc1) that the linter made after a merge from master from some reason. Should be safe to ignore.

**Review by markstory — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/organization_mapping/serial.py
Comment (GabeVillalobos) [reply src/sentry/services/hybrid_cloud/organization_mapping/serial.py]: Yes, organization mapping will no longer have a `verified` property or column in the database very soon. Removing it proactively from here should be fine.

**Review by GabeVillalobos — APPROVED** [review]

**Comment by corps** [comment]
Alright, this should now just work, lets gooooo

**Pull request merged**
